### PR TITLE
Implement service to handle favorites

### LINKED ISSUE #91: Add service to manage favourites.
- [x] use the #90 to read and write favourites
- [x] Provide a react context to provide the service
- [x] The context should contain the favorites so that the change detection works

### PR DESCRIPTION
**Pull request by Paulomart**
### Description

Closes #91

## Checklist

- [x] My code follows the style guidelines
- [x] I have documented my code
- [x] My changes generate no new warnings
- [x] I have linked the Issue
- [x] The Pull Request covers all the Checkpoints from the tagged Issue
- [x] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [x] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [ ] No tests required
- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by Paulomart** [comment]
@timvahlbrock this fails because an OEM error:

```
⎯⎯⎯⎯⎯⎯ Unhandled Errors ⎯⎯⎯⎯⎯⎯

Vitest caught 1 unhandled error during the test run.
This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.

⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯
Error: Worker terminated due to reaching memory limit: JS heap out of memory
 ❯ new NodeError node:internal/errors:405:5
 ❯ [kOnExit] node:internal/worker:287:26
 ❯ Worker.<computed>.onexit node:internal/worker:209:20

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: {
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
  "code": "ERR_WORKER_OUT_OF_MEMORY",
}
```

I already increased the memory via `NODE_OPTIONS=--max-old-space-size=8192`, but this didnt seem to help.

**Comment by Paulomart** [comment]
@timvahlbrock whooops that was my fault. Had a bad test...

**Comment by timvahlbrock** [comment]
favourite is British English. Are we bothered by that?

**Comment by Paulomart** [comment]
@Paulomart aghr ups. But I feel the change isn't worth the time...

File: frontend/src/services/persistence/generatePersistedObjectStorage.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        persistence,
+        setPersistedObjects,
+        stringifyPersistedObjects,
+        persistenceKey
+      ]
+    );
+
+    const addPersistedObject = useCallback(
+      (createPersistedObject: TCreateObject): PersistedObject<TCreateObject> => {
+        console.log("addPersistedObject", createPersistedObject);
```
Comment (timvahlbrock) [inline frontend/src/services/persistence/generatePersistedObjectStorage.tsx:57]: Debug statements in prod code

**Review by timvahlbrock — CHANGES_REQUESTED** [review]

**Comment by Paulomart** [comment]
> And another thing: Would we reduce much of the effort if we used https://github.com/ionic-team/ionic-storage ?

I would be possible but that would have effected #225. This builds on top of that.

File: frontend/src/services/persistence/generatePersistedObjectStorage.tsx (hunk 1 above)
Comment (Paulomart) [reply frontend/src/services/persistence/generatePersistedObjectStorage.tsx]: Done

**Pull request merged**